pkg/controller/backup_session.go

```diff
@@ -242,12 +241,13 @@ func (c *StashController) ensureBackupJob(backupSession *api_v1beta1.BackupSessi
 	_, _, err = batch_util.CreateOrPatchJob(c.kubeClient, jobMeta, func(in *batchv1.Job) *batchv1.Job {
 		// set BackupSession as owner of this Job
 		core_util.EnsureOwnerReference(&in.ObjectMeta, backupConfigRef)
-		in.Labels = map[string]string{
-			// job controller should not delete this job on completion
-			// use a different label than v1alpha1 job labels to skip deletion from job controller
-			// TODO: Remove job controller, cleanup backup-session periodically
-			util.LabelApp: util.AppLabelStashV1Beta1,
+		if in.Labels == nil {
+			in.Labels = make(map[string]string)
 		}
+		// backup job is created by resolving task and function. we should not delete it when it goes to completed state.
+		// user might need to know what was the final resolved job specification for debugging purpose.
+		in.Labels[apis.KeyDeleteJobOnCompletion] = "false"
```

Comment:
- Why is this in Label instead of Annotation?
- This sounds like should be a spec.reclaimPolicy so user can tell us what they want to do. 
- Or this can be a automatic - delete if backup is successful / delete otherwise.
- Who/how is the clean up done?

Reply:
TODO: https://github.com/stashed/stash/issues/807